prepareShapesAndCards.py

```diff
@@ -530,8 +546,25 @@ def prepareShapes(backgrounds, signals, discriminant, discriminantName):
     for signal in signals :
         cb = ch.CombineHarvester()
         cb.AddObservations(['*'], [''], ['_%s'%options.dataYear], [''], discriminant)
-        cb.AddProcesses(['*'], [''], ['_%s'%options.dataYear], [''], backgrounds, discriminant, False)
         cb.AddProcesses(['*'], [''], ['_%s'%options.dataYear], [''], [signal], discriminant, True)
+        if options.dataYear == '2016':
+            cb.AddProcesses(['*'], [''], ['_%s'%options.dataYear], [''], backgrounds, discriminant, False)
+        else:
+            if not 'b2j3' in discriminantName:
+                try: backgrounds.remove('qcd')
+                except: pass
+            else:
+                if not 'qcd' in backgrounds: backgrounds.append('qcd')
+            if 'all' in discriminantName:
+                if signal == 'Hut':
+                    discriminant.remove((1, 'DNN_Hut_b2j3'))
```

Comment:
Can you explain this?

Reply:
For QCD, remove b2j3 first and add them in the next line, including qcd process. Unless, the code will add b2j3 without qcd in background processes. Yes, there is a danger of missing something while exclude qcd for b2j3 or include more category. But it is hard to organize existing code for that kind of task, because code fragments are shattered in several different files..